### PR TITLE
Modifying the status command to receive a directory path as an argument

### PR DESCRIPTION
**Pull request by Mun1z**
**Obs: Depends of PR #48**

Directory path argument on status command to show only files inside directory. 
Example:  `ml-git dataset status imagenet8`
```
INFO - Repository: dataset: status of ml-git index for [imagenet8]
Changes to be committed:
Untracked files:
        README.md
        imagenet8.spec
        data\file1.jpg
Corrupted files:
```
with argument  on command: `ml-git dataset status imagenet8 data`

```
INFO - Repository: dataset: status of ml-git index for [imagenet8]
Changes to be committed:
Untracked files:
        file1.jpg
 Corrupted files:
```

Demonstration video: https://asciinema.org/a/RZ7cEbuh7yi40PonswxzDy9yM

### CODE REVIEW

File: tests/integration/test_36_status_path_directory.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    @pytest.mark.usefixtures('start_local_git_server', 'switch_to_tmp_dir')
+    def test_03_status_after_put_more_than_one_file_in_dataset_with_directory(self):
+        self.set_up_status('dataset')
+        data_path = os.path.join(self.tmp_dir, 'dataset', 'dataset-ex', 'data')
+        os.makedirs(data_path, exist_ok=True)
+        create_file(data_path, 'file3', '0', '')
+        create_file(data_path, 'file4', '0', '')
+        self.assertRegex(check_output(MLGIT_STATUS_DIRECTORY % ('dataset', 'dataset-ex', 'data')),
+                         r'Changes to be committed:\s+Untracked files:(\s|.)*file3(\s|.)*file4(\s|.)')
```
Comment (lucasncabral) [inline tests/integration/test_36_status_path_directory.py:48]: Could you create more test scenarios for the command? For example:

- Status passing a directory that does not exist.
- Status of a directory after executing the ml-git add command

**Review by lucasncabral — CHANGES_REQUESTED** [review]

File: ml_git/file_system/local.py [hunk 2]
```diff
@@ -642,7 +642,7 @@ def remote_fsck(self, metadata_path, tag, spec_file, retries=2, thorough=False,
         return True
 
     def exist_local_changes(self, spec_name):
-        new_files, deleted_files, untracked_files, _, _ = self.status(spec_name, log_errors=False)
+        new_files, deleted_files, untracked_files, _, _ = self.status(spec_name, status_directory="", log_errors=False)
```
[the remark this answers is not in the document]
Comment (Mun1z) [reply ml_git/file_system/local.py]: Done.

File: docs/mlgit_commands.md [hunk 3]
```diff
@@ -593,7 +593,8 @@ Usage: ml-git dataset status [OPTIONS] ML_ENTITY_NAME
   index/staging area.
 
 Options:
-  --help  Show this message and exit.
+  --full     Show all contents for each directory.
+  --verbose  Debug mode
 ```
 
 Example:
```
[the remark this answers is not in the document]
Comment (Mun1z) [reply docs/mlgit_commands.md:600]: Done.

File: tests/integration/test_36_status_path_directory.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import unittest
+
+import pytest
+
+from tests.integration.commands import MLGIT_STATUS_DIRECTORY
+from tests.integration.helper import check_output, init_repository, create_file
+
+
+@pytest.mark.usefixtures('tmp_dir')
+class StatusShortModeAcceptanceTests(unittest.TestCase):
```
[the remark this answers is not in the document]
Comment (Mun1z) [reply tests/integration/test_36_status_path_directory.py]: Done.

File: ml_git/file_system/local.py [hunk 5]
```diff
@@ -685,7 +685,7 @@ def get_corrupted_files(self, spec):
 
         return corrupted_files
 
-    def status(self, spec, log_errors=True):
+    def status(self, spec, status_directory="", log_errors=True):
```
[the remark this answers is not in the document]
Comment (Mun1z) [reply ml_git/file_system/local.py]: Done on commit [1f90927](https://github.com/HPInc/ml-git/pull/50/commits/1f90927a4a232892ab9ad9670ce8e91e24096d64)

File: tests/integration/test_36_status_path_directory.py (hunk 1 above)
Comment (Mun1z) [reply tests/integration/test_36_status_path_directory.py]: Done.

File: tests/integration/test_35_status_short_mode.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+    @pytest.mark.usefixtures('start_local_git_server', 'switch_to_tmp_dir')
+    def test_06_status_after_delete_file(self):
+        self.set_up_checkout('dataset')
+        self.assertNotIn(ERROR_MESSAGE, check_output(MLGIT_CHECKOUT % ('dataset',
+                                                                       'computer-vision__images__dataset-ex__1')))
+        data_path = os.path.join(self.tmp_dir, 'dataset', 'computer-vision', 'images', 'dataset-ex', 'data')
+        file_to_be_deleted = os.path.join(data_path, 'file')
+        file_to_be_deleted2 = os.path.join(data_path, 'file2')
+        os.chmod(file_to_be_deleted, S_IWUSR | S_IREAD)
```
Comment (gilMars) [inline tests/integration/test_35_status_short_mode.py:107]: Use clear function in defined in helper.py

Comment (Mun1z) [reply tests/integration/test_35_status_short_mode.py]: Done on commit [529de97](https://github.com/HPInc/ml-git/pull/50/commits/529de9770f269d9784f919fa0d2c7b1bffd61e89)

**Review by andersonpablosilva — APPROVED** [review]

**Review by gilMars — APPROVED** [review]

**Review by lucasncabral — APPROVED** [review]

**Review by diego-pedroso — APPROVED** [review]

**Pull request merged**